File: Android Dev/TourGuide/app/src/main/java/com/example/chandigarhtourguide/MainActivity.java
package com.example.chandigarhtourguide;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
public CardView hotel, restaurant, places, shopping, museum, education;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //defining cards
        setContentView(R.layout.activity_main);
        hotel=findViewById(R.id.hotel);
        restaurant=findViewById(R.id.restaurant);
        places=findViewById(R.id.places);
        shopping=findViewById(R.id.shopping);
        museum=findViewById(R.id.museum);
        education=findViewById(R.id.education);

        //adding on click listener to the cards
        hotel.setOnClickListener(this);
        restaurant.setOnClickListener(this);
        places.setOnClickListener(this);
        shopping.setOnClickListener(this);
        museum.setOnClickListener(this);
        education.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        Intent i;
        switch (v.getId()){
            case R.id.hotel:i=new Intent(this, Hotels.class);
            startActivity(i); break;
            case R.id.restaurant:i=new Intent(this, Restaurants.class);
            startActivity(i); break;
            case R.id.places:i=new Intent(this, Places.class);
            startActivity(i);break;
            case R.id.shopping:i=new Intent(this, Shopping.class);
            startActivity(i); break;
            case R.id.museum:i=new Intent(this, Museum.class);
            startActivity(i); break;
            case R.id.education:i=new Intent(this, Amusement.class);
            startActivity(i);break;
            default:break;
        }

    }
}

File: kotlin/Meme Share/app/src/main/java/com/example/memeshare/MainActivity.kt
package com.example.memeshare

import android.content.Intent
import android.graphics.drawable.Drawable
import android.media.Image
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.Toast
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import com.bumptech.glide.Glide
import com.bumptech.glide.load.DataSource
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.target.Target

class MainActivity : AppCompatActivity() {

    var currentImageUrl : String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        loadMeme()


    }

    private fun loadMeme(){
        val progressBar = findViewById<ProgressBar>(R.id.progressBar)
        progressBar.visibility = View.VISIBLE

        // Instantiate the RequestQueue.
        val queue = Volley.newRequestQueue(this)
        val url = "https://meme-api.herokuapp.com/gimme"

        // Request a string response from the provided URL.
        val jsonObjectRequest = JsonObjectRequest(
            Request.Method.GET, url, null,
            { response ->
                val memeImageView = findViewById<ImageView>(R.id.memeImageView)
                val url = response.getString("url")
                currentImageUrl = url

                Glide.with(this).load(url).listener(object : RequestListener<Drawable>{

                    override fun onLoadFailed(
                        e: GlideException?,
                        model: Any?,
                        target: Target<Drawable>?,
                        isFirstResource: Boolean
                    ): Boolean {

                        progressBar.visibility = View.GONE
                        return false

                    }

                    override fun onResourceReady(
                        resource: Drawable?,
                        model: Any?,
                        target: Target<Drawable>?,
                        dataSource: DataSource?,
                        isFirstResource: Boolean
                    ): Boolean {
                        progressBar.visibility = View.GONE
                        return false

                    }


                }).into(memeImageView)
            },
            {
                Toast.makeText(this, "Error Occurred", Toast.LENGTH_LONG).show()
            })

        // Add the request to the RequestQueue.
        queue.add(jsonObjectRequest)



    }

    fun shareMeme(view: View) {
//        for sharing any app we use INTENT and use the send action to share the files

        val intent = Intent(Intent.ACTION_SEND)
        intent.type = "text/plain"
        intent.putExtra(Intent.EXTRA_TEXT, "Hey checkout this cool meme at $currentImageUrl")

//        now we'll create a choose to choose which app we want to share to may be WP, insta, copytoclipboard
        val chooser = Intent.createChooser(intent, "Share this meme using...")
        startActivity(chooser)

    }
    fun nextMeme(view: View) {
        loadMeme()
    }
}